src/transaction.ts

```diff
@@ -119,5 +118,5 @@ function areAmountsCorrect({
     side === SwapSide.SELL ? queryParams.srcAmount : queryParams.destAmount;
   const priceRouteAmount =
     side === SwapSide.SELL ? priceRoute.srcAmount : priceRoute.destAmount;
-  return new BigNumber(priceRouteAmount).isEqualTo(amount);
+  return BigInt(priceRouteAmount) === BigInt(amount);
```

Comment:
Since the amounts start as string, we could probably get away with comparing them directly
And BigInt can't handle decimals: `BigInt('3.0') and BigInt('1e18') throw SyntaxError`, `BigInt(3.2) throws RangeError`, though `BigInt(3.0)` works.

Though here we only should have amounts in wei, so no danger of errors.

But in that case there's not much use in BigInt here, since it won't even preformat strings. Let's use `priceRouteAmount === amount`

Reply:
and you're right